crates/circuit/src/dag_circuit.rs

```diff
@@ -5731,52 +6414,54 @@ impl DAGCircuit {
-                                vars.push(var);
-                            }
-                        }
-                    }
+        if let Some(instr) = self.try_view_control_flow_inner(instr) {
+            match instr {
+                ControlFlowView::IfElse {
+                    condition: Condition::Expr(condition),
+                    ..
                 }
-
-                // TODO: this is the Python-side `ControlFlowOp.iter_captured_vars` which iterates
-                //   over vars in all blocks of the op. This needs to be ported to Rust when control
-                //   flow is ported.
-                for var in op.call_method0("iter_captured_vars")?.try_iter()? {
-                    vars.push(self.vars.find(&var?.extract()?).unwrap())
+                | ControlFlowView::While {
+                    condition: Condition::Expr(condition),
+                    ..
+                } => {
+                    // TODO: do we need to add bits from non-expr condition?
+                    let (expr_clbits, expr_vars) = wires_from_expr(condition)?;
+                    for bit in expr_clbits {
+                        clbits.push(bit);
+                    }
+                    for var in expr_vars {
+                        vars.push(var);
+                    }
                 }
-                if op.is_instance(imports::SWITCH_CASE_OP.get_bound(py))? {
-                    let target = op.getattr(intern!(py, "target"))?;
-                    if target.cast::<PyClbit>().is_ok() {
-                        let target_clbit: ShareableClbit = target.extract()?;
-                        clbits.push(self.clbits.find(&target_clbit).unwrap());
-                    } else if target.is_instance_of::<PyClassicalRegister>() {
-                        for bit in target.try_iter()? {
-                            let clbit: ShareableClbit = bit?.extract()?;
-                            clbits.push(self.clbits.find(&clbit).unwrap());
+                ControlFlowView::Switch { target, .. } => match target {
+                    SwitchTarget::Bit(bit) => {
+                        clbits.push(self.clbits.find(bit).unwrap());
+                    }
+                    SwitchTarget::Register(reg) => {
+                        for bit in reg.bits() {
+                            clbits.push(self.clbits.find(&bit).unwrap());
                         }
-                    } else {
-                        let (expr_clbits, expr_vars) = wires_from_expr(&target.extract()?)?;
+                    }
+                    SwitchTarget::Expr(expr) => {
+                        let (expr_clbits, expr_vars) = wires_from_expr(expr)?;
                         for bit in expr_clbits {
                             clbits.push(bit);
                         }
                         for var in expr_vars {
                             vars.push(var);
                         }
                     }
+                },
+                _ => (),
```

Comment:
Please can we avoid the `_` patterns - I can't actually tell if this is valid or not because I can't see which conditions are being ignored.

Reply:
Done in some previous commit, tweaked in 0bad00dae4.